review/get_review_ids.py

```diff
@@ -0,0 +1,212 @@
+
+    enroll_user(user, current_course)
+    
+    problem_data = []
+    # Each record corresponds to a problem the user has loaded
+    # in the original course
+    for record in StudentModule.objects.filter(**{'student_id': user.id, 'course_id': current_course, 'module_type': 'problem'}):
+        # record.module_state_key takes the form: 
+        # block-v1:{course_id}+type@problem+block@{problem_id}
+        problem_id = str(record.module_state_key).split("@")[-1]
+
+        # Actual logic regarding the record should go here
+        state = json.loads(record.state)
+
+        # The key 'selected' shows up if a problem comes from a
+        # library content module. These cause issues so we skip this.
+        # Issue: Library content contains problems but the CSM brings up
+        # the library content and not the problems within
+        if 'selected' in state:
+            continue
+
+        correctness, attempts = get_correctness_and_attempts(state)
+
+        problem_data.append((problem_id, correctness, attempts))
+
+    if len(problem_data) < num_desired:
+        return []
+
+    problems_to_show = random.sample(problem_data, num_desired)
+    review_course_id = REVIEW_COURSE_MAPPING[str(current_course)]
+    urls = []
+    for problem, correctness, attempts in problems_to_show:
+        urls.append((TEMPLATE_URL.format(course_id=review_course_id,
+                    type='problem', xblock_id=problem), correctness, attempts))
+    return urls
+
+def get_vertical(current_course):
+    '''
+    Looks through all the problems a learner has previously loaded and
+    finds their parent vertical. Then randomly selects a single vertical
+    to show the learner.
+
+    Parameters:
+        current_course (CourseLocator): The course the learner is currently in
+
+    Returns a url (str) with the vertical id to render for review.
+    '''
+    user = crum.get_current_user()
+
+    enroll_user(user, current_course)
+
+    store = modulestore()
+    course_usage_key = store.make_course_usage_key(current_course)
+    course_blocks = get_course_blocks(user, course_usage_key)
+
+    vertical_data = set()
+
+    # Each record corresponds to a problem the user has loaded
+    # in the original course
+    for record in StudentModule.objects.filter(**{'student_id': user.id, 'course_id': current_course, 'module_type': 'problem'}):
+        # record.module_state_key takes the form:
+        # block-v1:{course_id}+type@problem+block@{problem_id}
+        problem_id = str(record.module_state_key).split("@")[-1]
+
+        # Actual logic regarding the record should go here
+        state = json.loads(record.state)
+
+        # The key 'selected' shows up if a problem comes from a
+        # library content module. These cause issues so we skip this.
+        # Issue: Library content contains problems but the CSM brings up
+        # the library content and not the problems within
+        if 'selected' in state:
+            continue
```

Comment:
This logic seems the same as above. You should implement a Python generator (using `yield`) to implement this common logic.